tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailbox.scala

```diff
@@ -0,0 +1,26 @@
+package com.linagora.tmail.team
+
+import org.apache.james.core.Username
+import org.apache.james.mailbox.model.{MailboxConstants, MailboxPath}
+
+object Namespace {
+  val TEAM_MAILBOX_NAMESPACE: Namespace = Namespace(MailboxConstants.NAMESPACE_PREFIX_CHAR + "tm")
+}
+
+case class Namespace(value: String) extends AnyVal
+
+case class MailboxName(value: String) extends AnyVal
+
+object TeamMailbox {
+  def from(mailboxPath: MailboxPath): TeamMailbox =
+    TeamMailbox(
+      Namespace(mailboxPath.getNamespace),
+      mailboxPath.getUser,
+      MailboxName(mailboxPath.getName))
+}
+
+case class TeamMailbox(namespace: Namespace, tmUser: Username, mailboxName: MailboxName) {
+  def mailboxPath: MailboxPath = new MailboxPath(namespace.value, tmUser, mailboxName.value)
+
+  def asString: String = s"${namespace.value}:${tmUser.getDomainPart.orElseThrow().asString()}:${tmUser.getLocalPart}.${mailboxName.value}"
```

Comment:
IMO `TeamMailbox` should represent the parent mailbox of the `Sent` and `Inbox`, so given a Team mailbox it makes sense to ask:

```
def inbox: MailboxPath = ???
def sent: MailboxPath = ???
```